### PR TITLE
Sample app redesign [Part 1]

### PR DESCRIPTION
**Pull request by AndreiMisiukevich**
### Description of Change ###

- Redesigned sample app;
- Cleaned and refactored code;
- Made XF version consistent across all solution projects;
- Added a new "folder" - **Extensions** with single MarkupExtension that helps to work with local images.

**Note**: this is the first step of the sample app redesign.
Once this PR is merged, I will submit a follow-up PR with updates within #132 

### Bugs Fixed ###
- Related to issue #129 

### API Changes ###

Added: 
- `ImageResourceExtension`

### Behavioral Changes ###
Jun-18-2020 03-49-01

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [X] Rebased on top of master at time of PR
- [X] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

File: XamarinCommunityToolkitSample/App.xaml [hunk 1]
```diff
@@ -4,8 +4,47 @@
...
+        </Style>
+
+        <Style TargetType="ContentPage" ApplyToDerivedTypes="True">
+            <Setter Property="BackgroundColor" Value="{StaticResource AppBackgroundColor}" />
+            <Setter Property="ios:Page.UseSafeArea" Value="True" />
+        </Style>
+
+        <Style TargetType="Button">
+            <Setter Property="FontSize" Value="Medium" />
+            <Setter Property="BackgroundColor" Value="#1976D2" />
```
Comment (Depechie) [inline XamarinCommunityToolkitSample/App.xaml:33]: Maybe also add this as static resource Color.

File: XamarinCommunityToolkitSample/Views/WelcomePage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            => id switch
+            {
+                Section.SectionId.Behaviors => new BehaviorsPage(),
+                Section.SectionId.Converters => new ContentPage { Title = "Not Implemented yet" },
+                Section.SectionId.Views => new ContentPage { Title = "Not Implemented yet" },
+                Section.SectionId.TestCases => new ContentPage { Title = "Not Implemented yet" },
+                _ => throw new System.NotImplementedException()
+            };
+
+        public sealed class Section
```
Comment (Depechie) [inline XamarinCommunityToolkitSample/Views/WelcomePage.xaml.cs:34]: Classes best in a separate file.

File: XamarinCommunityToolkitSample/ViewModels/BaseViewModel.cs [hunk 3]
```diff
@@ -1,56 +1,24 @@
...
-
-            changed.Invoke(this, new PropertyChangedEventArgs(propertyName));
-        }
-
-        #endregion
+        protected virtual void OnPropertyChanged([CallerMemberName] string name = null)
+            => PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
     }
-}
+}
```
Comment (Depechie) [inline XamarinCommunityToolkitSample/ViewModels/BaseViewModel.cs:24]: I think the BaseViewModel is no longer used? Use it, or drop it I guess.

File: XamarinCommunityToolkitSample/Views/WelcomePage.xaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    </NavigationPage.TitleView>
+
+    <StackLayout>
+        <Image Source="{extensions:ImageResource xamarin_monkey.png}" HeightRequest="250" />
+        <ScrollView Padding="{StaticResource ContentPadding}"
+                    VerticalOptions="FillAndExpand">
+            <StackLayout Spacing="40">
+                <Label Text="Welcome to XamarinCommunityToolkit" FontSize="Large" />
+                <StackLayout Spacing="10"
+                             BindableLayout.ItemsSource="{Binding Sections, Source={x:Reference Page}}">
```
Comment (Depechie) [inline XamarinCommunityToolkitSample/Views/WelcomePage.xaml:22]: See remark about BaseViewModel...

File: XamarinCommunityToolkitSample/Views/WelcomePage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        ICommand navigateCommand;
+
+        public WelcomePage()
+            => InitializeComponent();
+
+        public ICommand NavigateCommand => navigateCommand ??= new Command(parameter
+            => Navigation.PushAsync(GetPage((Section.SectionId)parameter)));
+
+        public Section[] Sections { get; } =
```
Comment (Depechie) [inline XamarinCommunityToolkitSample/Views/WelcomePage.xaml.cs:16]: See remark about BaseViewModel

File: XamarinCommunityToolkitSample/Views/WelcomePage.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                Section.SectionId.TestCases => new ContentPage { Title = "Not Implemented yet" },
+                _ => throw new System.NotImplementedException()
+            };
+
+        public sealed class Section
+        {
+            public SectionId Id { get; set; }
+            public string Text { get; set; }
+
+            public enum SectionId
```
Comment (Depechie) [inline XamarinCommunityToolkitSample/Views/WelcomePage.xaml.cs:39]: I would add enums in a separate Enums.cs file in Models folder.

**Comment by AndreiMisiukevich** [comment]
@Depechie the sample is updated

File: XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+namespace XamarinCommunityToolkitSample.Models
+{
+    public enum WelcomeSectionId
```
Comment (Depechie) [inline XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs:5]: I would still extract this to an Enums.cs file, not sure what others think of this though... @sthewissen  ?

File: XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        Behaviors,
+        Converters,
+        Extensions,
+        Views,
+        TestCases
+    }
+
+    public static class WelcomeSectionIdExtensions
+    {
```
Comment (Depechie) [inline XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs:15]: I would rename the file to include Extension

Comment (AndreiMisiukevich) [reply XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs:15]: Do you mean extract this class to the separate file?

Comment (Depechie) [reply XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs:15]: Yeah indeed

File: XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs (hunk 7 above)
Comment (AndreiMisiukevich) [reply XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs:5]: My opinion is that we should keep every class/struct/enum in the separate files.
But I'd like to make an exception for Enum extensions cuz that's the only option to add some methods there. (to comment below)

Comment (Depechie) [reply XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs:5]: Yeah I get it... still even in these cases I separated them. But like I said, not a big problem in this case, so if no other reviewer objects, current code can be kept.

File: XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs (hunk 8 above)
Comment (AndreiMisiukevich) [reply XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs:15]: Are you Ok with that as well?

That's like extension in swift :) or partial class in C#

File: XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs (hunk 7 above)
Comment (sthewissen) [reply XamarinCommunityToolkitSample/Models/WelcomeSectionId.cs:5]: I'm fine with things being their own class.

**Pull request merged**